src/Clock.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+class Clock extends React.Component {
+  state = {
+    date: new Date(),
+    name: 0,
+    visibility: false,
+  };
+
+  componentDidMount() {
+    setInterval(() => {
+      this.setState({ date: new Date() });
+      // eslint-disable-next-line no-console
+      console.log(this.state.date.toLocaleTimeString());
+    }, 1000);
+  }
+
+  changeName = () => {
+    this.setState((state) => {
+      const random = Math.floor(Math.random() * Math.floor(1000));
+
+      // eslint-disable-next-line no-console
+      console.log(`The Clock has been renamed from ${state.name} to ${random}`);
```

Comment:
try to move console.log into conditions when visibilty true, so it could disappear, when clocks hidden